### PR TITLE
Remove legacy prefixes, clean up asset names

### PR DESCRIPTION
**Pull request by elesel**
- Remove microsoft_azureml_ prefix
- Change underscores to dashes in asset names
- Clean up directory names
- Remove "component" asset name suffixes

### CODE REVIEW

File: training/automl/components/text_ner/spec.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 $schema: http://azureml/sdk-2-0/AutoMLComponent.json
-name: microsoft_azureml_automl_text_ner_component
+name: automl-text-ner
```
Comment (novaturient95) [inline training/automl/components/text_ner/spec.yaml:2]: @shouryaj can you please help to confirm naming of AutoML components. "-" will not work for registering component using CLI/SDK. But in general if we should get rid of **microsoft_azureml** prefix

Comment (elesel) [reply training/automl/components/text_ner/spec.yaml:2]: There's an ongoing discussion regarding dashes in component names in the AzureML Python SDK channel.

**Comment by skasturi** [comment]
Do we happen to have workflows that are executing these as BVTs?

**Comment by skasturi** [comment]
If yes, do we need to change those as well as i see assets & spec being changed? 

May be related to the infra work that @quchuyuan is doing? 

---
In reply to: [1378301920](https://github.com/Azure/azureml-assets/pull/317#issuecomment-1378301920) [](http://example.com/codeflow?ancestors=1378301920)

**Pull request closed without merging**